tests/loki/directbeam_test.py

```diff
@@ -32,7 +38,8 @@ def test_can_compute_direct_beam_for_all_pixels():
     )
     providers = loki_providers()
     pipeline = sciline.Pipeline(providers, params=params)
-    I0 = _get_I0(qbins=params[QBins])
+    pipeline.set_param_series(PixelMaskFilename, [])
+    I0 = _get_I0(qbins=params[QBins].edges['Q'])
```

Comment:
I can see that this is not optimal UX. But at the same time, I don't really know how else to do this. The `QBins` is serving a double purpose. It reminds me a little of the `WavelengthBins` vs `WavelengthBands` which are kind of similar but not really the same thing.

Are `QxyBins` and `QBins` actually mutually exclusive? I know that is what we said in the past. But could you ever want to make a plot of e.g. `Qx` vs `Q`?

Could another alternative be to have a parameter for each `QBins`, `QxBins`, `QyBins` and then you have to request a different result, e.g. `BackgroundSubtractedIofQ` vs `BackgroundSubtractedIofQxy`. And then you would have different providers that take in either `Qbins` or `QxBins` and `QyBins`?

Reply:
> Are `QxyBins` and `QBins` actually mutually exclusive? I know that is what we said in the past. But could you ever want to make a plot of e.g. `Qx` vs `Q`?

I don't think so, and catering to hypotheticals does not feel like a good choice?
 
> Could another alternative be to have a parameter for each `QBins`, `QxBins`, `QyBins` and then you have to request a different result, e.g. `BackgroundSubtractedIofQ` vs `BackgroundSubtractedIofQxy`. And then you would have different providers that take in either `Qbins` or `QxBins` and `QyBins`?

Hmmm, interesting idea, I wonder how many providers and types would need to be duplicated, or whether generics could be used.

Reply:
I like the generics because it looks like a mathematical function `I[Q]`.
Would we then have `CleanSummed[Qxy, SampleRun, Numerator]`?
Could we have a problem where we have too many params in a generic and things start becoming difficult? I can't think a particular example right now but I seem to remember having issues with that (maybe it was when also mixing with parameter tables?).
In any case, if it works, I think it could be nice?

Reply:
Are we happy to make the change to such generics now, alongside this change? It would represent a major workflow change (for the users).

Also, it will not work with the current release of Sciline, since we do not have a `ScopeThreeParams` helper.